spec/models/bottleneck_event_spec.rb

```diff
@@ -76,4 +76,37 @@
       expect(query).to match(/resource_type = 'EmsCluster'.*resource_type = 'Host'/)
     end
   end
+
+  describe ".last_created_on" do
+    context "returns the last created_on date" do
+      def expect_last_created_on_to_eq_resource_last_created_on(resource)
+        bottleneck_event = BottleneckEvent.create!(:resource => resource)
+        expect(described_class.last_created_on(resource)).to eq(bottleneck_event.reload.created_on)
+      end
+
+      specify "for a host_redhat resource" do
+        resource = FactoryGirl.create(:host_redhat)
+        expect_last_created_on_to_eq_resource_last_created_on(resource)
+      end
+
+      specify "for a host_vmware resource" do
+        resource = FactoryGirl.create(:host_vmware)
+        expect_last_created_on_to_eq_resource_last_created_on(resource)
+      end
+      specify "for a miq_enterprise resource" do
+        resource = FactoryGirl.create(:miq_enterprise)
+        expect_last_created_on_to_eq_resource_last_created_on(resource)
+      end
+
+      specify "for a ems_redhat resource" do
+        resource = FactoryGirl.create(:ems_redhat)
+        expect_last_created_on_to_eq_resource_last_created_on(resource)
+      end
+
+      specify "for a ems_cluster_openstack resource" do
+        resource = FactoryGirl.create(:ems_cluster_openstack)
+        expect_last_created_on_to_eq_resource_last_created_on(resource)
+      end
```

Comment:
@gtanzillo judging from https://github.com/ManageIQ/manageiq/blob/719790ec755e79836828afcc14463aa917b61bbe/app/models/bottleneck_event.rb#L113-L124 these resources are the only reasonable to test, right?